docs/guides/application-commands/context-menu-commands/receiving-context-menu-command-events.md

```diff
@@ -0,0 +1,43 @@
+# Receiving Context Menu events
+
+User commands and Message commands have their own unique objects returned. Different from Slash commands. To get the appropriate object returned, you can use a similar method to the slash commands.
+
+```cs
+client.InteractionCreated += InteractionCreatedHandler;
+
+...
+
+public async Task InteractionCreatedHandler(SocketInteraction arg)
+{
+	if ( arg.Type == InteractionType.ApplicationCommand)
+		Task.Run(() => ApplicationCommandHandler(arg));
+}
+
+public async Task ApplicationCommandHandler(SocketInteraction arg)
+{
+	var slashCommand = arg as SocketSlashCommand;
+	if(slashCommand != null)
+		Console.Writeline("Slash command received!")
+		
+	var userCommand = arg as SocketApplicationUserCommand;
```

Comment:
Maybe do a switch statement for the types as it looks a bit nicer and is easier to read

Reply:
Unsure how to proceed with this.

Reply:
```
switch (arg) {
  case SocketApplicationUserCommand userCommand:
    // ...
    break;
  // ...
}
```

Reply:
i.... did not know this was a thing, honestly. So, it will only work if the cast results in non-null?

Reply:
It will check the type and if it matches cast to it, then it will be accessible as `userCommand` local property.

Also `if (smth is SomeType someType)` is useful instead of as and then checking for null, but if there are more types, switch is better